### PR TITLE
code.org/starwars: don't show JS tutorial for mobile/ios devices.

### PR DESCRIPTION
**Pull request by breville**
Code.org/starwars only shows the JS tutorial link if it determines that the user is neither on one of our "mobile" devices nor on iOS.

We also now specify some min-height values so that the layout is closer to correct before the main background image dimensions are determined.

Also factored out the duplicated mobilecheck code into a single partial, cleaned up its usage, and added more mobile devices, notably iPad.

### CODE REVIEW

**Comment by breville** [comment]
fyi @kozzi

File: pegasus/sites.v3/code.org/public/starwars.haml [hunk 1]
```diff
@@ -55,3 +55,18 @@ social:
...
+
+:javascript
+
+  $(document).ready(function() {
+    window.isIOS = function() {
+      var reg = /iP(hone|od|ad)/;
+      return reg.test(window.navigator.userAgent);
+    };
+
+    if (! window.isIOS() && ! window.mobilecheck()) {
```
Comment (tanyaparker) [inline pegasus/sites.v3/code.org/public/starwars.haml]: Do we have a styleguide for js? I usually would do `if (!window.isIOS() && !window.mobilecheck())`

**Comment by tanyaparker** [comment]
Including an image for reference of what it looks like on iPad

image

File: pegasus/sites.v3/code.org/views/learn_code_studio.haml [hunk 2]
```diff
@@ -120,12 +123,6 @@
...
-    return check; }
-
   // If mobile Star Wars block goes to blockly version
   if (window.mobilecheck()) {
```
Comment (tanyaparker) [inline pegasus/sites.v3/code.org/views/learn_code_studio.haml:125]: I feel like this one and the mobile check in learn_carousel_block should also include the iOS check. No? What if we made it a pegasus page_helper to check for any non-desktop device?

Comment (tanyaparker) [inline pegasus/sites.v3/code.org/views/learn_code_studio.haml:125]: Er-- can you use a ruby helper in the js?

I just wanted to simplify having 2 different kinds of "mobile" checks that include or not include iPad type devices. But maybe we should have two different views for it.

**Comment by tanyaparker** [comment]
LGTM

**Comment by tanyaparker** [comment]
Todo: My follow up PR will be to fix the code studio carousel on ipad & ipad mini

**Pull request merged**